WeatherApp/ViewModels/WeatherDetailViewModel.swift

```diff
@@ -29,34 +34,66 @@ class WeatherDetailViewModel {
         self.coordinator = coordinator
         self.locationService = appDependencies.locationService
         self.dataRepository = appDependencies.dataRepository
-        self.weeklyWeather = WeeklyWeather(city: currentWeather.city, dailyWeatherList: [], hourlyWeatherList: [])
+        self.weeklyWeather = BehaviorRelay(value: WeeklyWeather(city: currentWeather.city, dailyWeatherList: [], hourlyWeatherList: []))
+        
+        locationService.getLocationCoordinates(location: currentWeather.city)
+        getWeeklyWeather()
```

Comment:
Since you're already observing refreshData, no need for this call (or the `getWeeklyWeather()`). You can just trigger refreshData event.